### DISCUSSION TITLE
Default providers disabled by default - how to re-enable python provider

### DISCUSSION THREAD

**Discussion by ustrucx** [NvChad/NvChad#2212] [General]
Hi, first of all NvChad got me into neovim, this is an amazing project, thank you!

Now, after some time trying to understand why my python provider became disabled once I installed NvChad I found this in core/init.lua:

`-- disable some default providers
for _, provider in ipairs { "node", "perl", "python3", "ruby" } do
  vim.g["loaded_" .. provider .. "_provider"] = 0
end`

I have tried adding 'vim.g.python3_host_prog = '/usr/bin/python3' to custom/init.lua but it does nothing, it only works when I comment the code that disables it on core/init.lua but since editing that file is not advised, how do I re-enable what that code disabled? Thanks!!

**Comment by antoineco** [comment]
If this provider is needed by a specific plugin, I would advise enabling it in the plugin's `init = function()`, inside your `custom/plugins.lua`.

This approach also has the benefit to better communicate the intent behind enabling the provider (i.e. for which plugin it is required).

**Reply by siduck** [reply]
https://github.com/NvChad/NvChad/issues/1417

**Reply by ustrucx** [reply]
Thank you so much for the quick responses!!

I would not know how to proceed in practice with the suggested solution, perhaps something simple I haven't realized, but following the above link lead me to a person who mentioned the setting "vim.g.loaded_python3_provider = nil", I added it to my custom/init.lua and it worked. Thanks again!
